tests/csm/rest/test_iam_users.py

```diff
@@ -424,14 +424,233 @@ def test_35930(self):
+            buckets = s3_obj.bucket_list()[1]
+            assert_utils.assert_false(len(buckets), "buckets found on new IAM user")
+            self.log.info("Step: Verified no bucket present in new account")
+            self.log.info("Create bucket and perform IO")
+            self.log.info("Step: Verify create bucket")
+            status, resp = s3_obj.create_bucket(bucket_name)
+            assert_utils.assert_true(status, resp)
+            resp = s3_obj.bucket_list()
+            assert_utils.assert_in(bucket_name, resp[1], resp[1])
+            self.log.info("Step: Verified create bucket")
+            self.log.info("Step: Check bucket is empty")
+            resp = s3_obj.object_list(bucket_name)
+            assert resp[0], resp[1]
+            assert_utils.assert_true(resp[0], resp[1])
+            resp_bkt_lst = None if not resp[1] else resp[1]
+            # For checking the object list should be none
+            assert resp_bkt_lst is None, resp
+            assert_utils.assert_is_not_none(resp[0], resp[1])
+            self.log.info("Step: Verified that bucket was empty")
+            self.log.info("Step: Verify put object.")
+            test_file = "test-object.txt"
+            file_path_upload = os.path.join(TEST_DATA_FOLDER, test_file)
+            if os.path.exists(file_path_upload):
+                os.remove(file_path_upload)
+            if not os.path.isdir(TEST_DATA_FOLDER):
+                self.log.debug("File path not exists, create a directory")
+                system_utils.execute_cmd(cmd=common_cmd.CMD_MKDIR.format(TEST_DATA_FOLDER))
+            system_utils.create_file(file_path_upload, self.file_size)
+            resp = s3_obj.put_object(bucket_name=bucket_name, object_name=test_file,
+                                     file_path=file_path_upload)
+            self.log.info("Removing uploaded object from a local path.")
+            os.remove(file_path_upload)
+            assert_utils.assert_true(resp[0], resp[1])
+            self.log.info("Step: Verify get object.")
+            resp = s3_obj.get_object(bucket_name, test_file)
+            assert_utils.assert_true(resp[0], resp)
+        self.log.info("[END]Creating IAM users with different tenant")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    # pylint: disable=too-many-statements
+    # pylint: disable=too-many-locals
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.s3_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-39410')
+    def test_39410(self):
+        """
+        S3 IAM User: Create same name buckets with IAM users of same UID & name in different tenant
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info(
+            "[START] Create same name buckets with same name IAM users in different tenant")
+        bucket_name = "iam-user-bucket-" + str(int(time.time()))
```

Comment:
parameters from line no 586 to 588 are commonly used so better to keep them in setup_method to avoid code duplication.

Reply:
done